### PR TITLE
[choreolib] Rename `AutoLoop` to `AutoRoutine`

### PR DESCRIPTION
**Pull request by oh-yes-0-fps**
Renamed `AutoLoop` to `AutoRoutine`
Fixed `auto-routines` doc examples after removal of `TriggerExt`

### CODE REVIEW

**Comment by calcmogul** [comment]
Needs a C++ port.

**Comment by oh-yes-0-fps** [comment]
Cpp would likely just store a `std::string_view` in the routine to do trajectory equality with most recent run in routine.

**Comment by calcmogul** [comment]
std::string_view isn't an owning container, so it shouldn't be used as a member variable.

**Comment by spacey-sooty** [comment]
Can you split AutoLoop -> AutoRoutine to a seperate PR?

File: choreolib/src/main/java/choreo/auto/AutoTrajectory.java [hunk 1]
```diff
@@ -506,4 +506,13 @@ public Pose2d[] collectEventPoses(String eventName) {
...
     return poses;
   }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj instanceof AutoTrajectory) {
+      AutoTrajectory other = (AutoTrajectory) obj;
+      return this.name.equals(other.name);
+    }
+    return false;
```
Comment (spacey-sooty) [inline choreolib/src/main/java/choreo/auto/AutoTrajectory.java:516]: This can be simplified I believe

**Review by spacey-sooty — CHANGES_REQUESTED** [review]

**Review by spacey-sooty — APPROVED** [review]

**Pull request merged**